examples/splitnn/Dual_headed_SplitNN_Config2.ipynb

```diff
@@ -0,0 +1,359 @@
+    "            client_output[owner.id] = self.models[owner.id](data_pointer[owner.id].reshape([-1, 14*28]))\n",
+    "            remote_outputs.append(\n",
+    "                client_output[owner.id].move(self.server).requires_grad_()\n",
+    "            )\n",
+    "\n",
+    "        #concat outputs from all clients at server's location\n",
+    "        server_input = torch.cat(remote_outputs, 1)\n",
+    "\n",
+    "        #pass concatenated output from server's model segment\n",
+    "        server_output = self.models[\"server\"](server_input)\n",
+    "        \n",
+    "        # move to label_owner\n",
+    "        server_output = server_output.move(self.label_owner).requires_grad_()\n",
+    "        \n",
+    "        #pass through the back segment and return prediction\n",
+    "        pred = models[self.label_owner.id](server_output)\n",
+    "        \n",
+    "        return pred    \n",
+    "        \n",
+    "    def zero_grads(self):\n",
+    "        for opt in self.optimizers:\n",
+    "            opt.zero_grad()\n",
+    "        \n",
+    "    def step(self):\n",
+    "        for opt in self.optimizers:\n",
+    "            opt.step()\n",
+    "    \n",
+    "    def train(self):\n",
+    "        for loc in self.models.keys():\n",
+    "            for i in range(len(self.models[loc])):\n",
+    "                self.models[loc][i].train()\n",
+    "    \n",
+    "    def eval(self):\n",
+    "        for loc in self.models.keys():\n",
+    "            for i in range(len(self.models[loc])):\n",
+    "                self.models[loc][i].eval()\n",
+    "            \n",
+    "#     @property\n",
+    "#     def location(self):\n",
+    "#         return self.models[0].location if self.models and len(self.models) else None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Data preprocessing\n",
+    "transform = transforms.Compose([transforms.ToTensor(),\n",
+    "                              transforms.Normalize((0.5,), (0.5,)),\n",
+    "                              ])\n",
+    "\n",
+    "trainset = datasets.MNIST('mnist', download=True, train=True, transform=transform)\n",
+    "trainloader = torch.utils.data.DataLoader(trainset, batch_size=64, shuffle=True)\n",
+    "\n",
+    "# create some workers\n",
+    "client_1 = sy.VirtualWorker(hook, id=\"client_1\")\n",
+    "client_2 = sy.VirtualWorker(hook, id=\"client_2\")\n",
+    "\n",
+    "server = sy.VirtualWorker(hook, id= \"server\") \n",
+    "label_owner = sy.VirtualWorker(hook, id=\"label_owner\")\n",
+    "# label_owner = client_1\n",
+    "\n",
+    "data_owners = (client_1, client_2)\n",
+    "model_locations = [client_1, client_2, server, label_owner]\n",
+    "\n",
+    "#Split each image and send one part to client_1, and other to client_2\n",
+    "distributed_trainloader = Distribute_MNIST(data_owners=data_owners, data_loader=trainloader)"
```

Comment:
For the case where images go to one worker and labels to another, we have implemented the `vertically_federate` method of a `PartitionedDataset` (https://github.com/OpenMined/PyVertical/blob/master/src/future/dataset.py#L219).
Could you change this to use that class?

Reply:
Actually, the image is split into two parts, that are sent to client_1 and client_2. and label_owner has labels.
(https://github.com/OpenMined/PyVertical/blob/5f3f9c1b071868d035d1e2d3538bf3e148eefa03/src/future/dataset.py#L193)
I don't think this supports this case.
Would you like me to write a new method for splitting the image in `PartitionedDataset` among two workers?

Reply:
yes please